cpu/sam0_common/periph/adc.c

```diff
@@ -76,7 +76,8 @@ static void _adc_poweroff(void)
 
 static int _adc_configure(adc_res_t res)
 {
-    assert(res >= ADC_RES_8BIT && res <= ADC_RES_12BIT);
+    assert((res == ADC_RES_8BIT) || (res == ADC_RES_10BIT) ||
+           (res == ADC_RES_12BIT));
```

Comment:
Personally, I prefer the explicit. The fact that using the comparators led to a bug in the first place is a code smell for me. If `add_res_t` were indeed a simple linear enum, then it would be plausible. But it doesn't. It maps the sam0 valves directly into it's number line and makes an assumption that the unsupported values (6bit, 14bit, and 16bit) are outside of the range. C doesn't have type safe enums, so we don't get any guarantee that someone wouldn't just stick whatever value they wanted there. In particular, when you boil all the indirection away... samd21 8 bit resolution is a (3 << 4), 12 bit is (0 << 4) and 10 bit is (2 << 4). There's a hole there (1 << 4), which happens to be 16 bit (averaging), which is currently not supported in riot. I also think that given a choice between code that explains itself, vs code that doesn't AND need comments to explain why it works a wonky way, the simple approach is better.